exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/internal/data/ImmutableAttributeUnitData.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.exporter.otlp.profiles.AttributeUnitData;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+@AutoValue
+public abstract class ImmutableAttributeUnitData implements AttributeUnitData {
```

Comment:
Can you add simple javadoc to all public classes and public methods?

Can you add the following boilerplate to all public internal classes?
```
 * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
 * at any time.
```
See [ImmutableMetricData](https://github.com/open-telemetry/opentelemetry-java/blob/22aaae8d441fb38479ace65ff449b5521d537983/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/ImmutableMetricData.java#L23-L28) for an example.